### PR TITLE
Add log4j2 hooks to other loggings

### PR DESCRIPTION
**Pull request by drcrallen**
As suggested by @housejester  in https://github.com/druid-io/druid/pull/1140

### CODE REVIEW

**Comment by drcrallen** [comment]
probably will need merge conflict resolution after https://github.com/druid-io/druid/pull/1140 goes in

File: docs/content/Configuration.md [hunk 1]
```diff
@@ -34,6 +34,12 @@ We recommend just setting the base ZK path and the ZK service host, but all ZK p
 
 See the [Zookeeper](ZooKeeper.html) page for more information on configuration options for ZK integration.
 
+### General Logging
+See [Logging](Logging.html) for more information
+|Property|Description|Default|Recommended|
+|--------|-----------|-------|-----------|
+|`java.util.logging.manager`|The logging manager class (this is a standard java config)| `null <<java default>>`|`org.apache.logging.log4j.jul.LogManager`
```
Comment (xvrl) [inline docs/content/Configuration.md]: this should not be part of the config docs, since this has to be provided as a system property and cannot be passed as part of the runtime.properties file.

Comment (drcrallen) [inline docs/content/Configuration.md]: Fixed. Moved to section higher in doc where server properties are defined.

**Comment by drcrallen** [comment]
There was some internal discussion on if the log4j2 log manager could have the same performance impact as doing an [slf4j bridge](http://www.slf4j.org/api/org/slf4j/bridge/SLF4JBridgeHandler.html)

I'm not convinced slf4j bridge for jul is the best option right now solely for the sake of arbitrary compatibility without some performance tests to make sure it doesn't tank logging performance.

File: pom.xml [hunk 2]
```diff
@@ -397,6 +398,20 @@
...
+              Please make sure to do performance tests before switching this to slf4j
+              -->
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-jul</artifactId>
+                <version>${log4j.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.slf4j</groupId>
+                <artifactId>jcl-over-slf4j</artifactId>
+                <version>${slf4j.version}</version>
```
Comment (housejester) [inline pom.xml]: Alternatively, you could also use log4j2's own jcl bridge: http://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-jcl
http://logging.apache.org/log4j/log4j-2.1/log4j-jcl/index.html
Since druid prefers to uses log4j2 config, then it might be better performance wise to use their bridges vs going from jcl->slf4j->log4j for example. Same could be said for the comment above about slf4j jul bridge. But I may be missing something in the underpinnings of log setup :)

Comment (drcrallen) [inline pom.xml]: The soft-goal is to reduce the dependency on log implementation, which is why most logging in Druid is slf4j based, and we just happen to use the log4j implementation. It *should* be directly pluggable with Logback (assuming proper config of course). The slf4j team has raised no concerns about performance of jcl-over-slf4j that I have seen, but were very explicit about performance problems in the jul bridge.

The log4j-jul breaks this general goal. Notably there is no note either way from the log4j2 folks regarding log4j-jul performance, which is concerning. If anyone has an example of performance differences that would be very helpful.

 Basically I would be proposing the slf4j jul bridge if it weren't for the performance concern.

File: pom.xml [hunk 3]
```diff
@@ -397,6 +398,20 @@
                 <version>${log4j.version}</version>
             </dependency>
             <dependency>
+              <!-- This is not slf4j's version because of performance concerns
+              http://www.slf4j.org/api/org/slf4j/bridge/SLF4JBridgeHandler.html
+              Please make sure to do performance tests before switching this to slf4j
```
Comment (xvrl) [inline pom.xml]: maybe also add a note that if someone wants to switch to slf4j they should use logback, which alleviates the problem that jul-to-slf4j has thanks to the [LevelChangePropagator](http://logback.qos.ch/manual/configuration.html#LevelChangePropagator)

Comment (drcrallen) [inline pom.xml]: Added note
